ddtrace/tracer/tracer.go

```diff
@@ -296,12 +353,14 @@ func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOpt
 
 // Stop stops the tracer.
 func (t *tracer) Stop() {
+	t.stopping.Lock()
+	defer t.stopping.Unlock()
```

Comment:
It's a bit gross how many fields we need only to shut down cleanly. I wish this wasn't here but I'm glad you've added it.